pyOCD/gdbserver/gdbserver.py

```diff
@@ -730,74 +775,75 @@ def handleQuery(self, msg):
-                        resp = 'Invalid Command: "%s"\n' % cmd
-                        resp = hexEncode(resp)
-                        return self.createRSPPacket(resp)
-                    else:
-                        resultMask = resultMask | safecmd[cmd_sub][1]
-                #if it's a single cmd, just launch it!
-                if len(cmdList) == 1:
-                    tmp = eval ('self.target.%s()' % cmd_sub)
-                    logging.debug(tmp)
-                    resp = "OK"
-                else:
-                    #10000001 for help reset, so output reset cmd help information
-                    if resultMask == 0x81:
-                        resp = 'Reset the target\n'
-                        resp = hexEncode(resp)
-                    #10000010 for help halt, so output halt cmd help information
-                    elif resultMask == 0x82:
-                        resp = 'Halt the target\n'
-                        resp = hexEncode(resp)
-                    #10000100 for help resume, so output resume cmd help information
-                    elif resultMask == 0x84:
-                        resp = 'Resume the target\n'
-                        resp = hexEncode(resp)
-                    #11 for reset halt cmd, so launch self.target.resetStopOnReset()
-                    elif resultMask == 0x3:
-                        resp = "OK"
-                        self.target.resetStopOnReset()
-                    #111 for reset halt resume cmd, so launch self.target.resetStopOnReset() and self.target.resume()
-                    elif resultMask == 0x7:
-                        resp = "OK"
-                        self.target.resetStopOnReset()
-                        self.target.resume()
-                    else:
-                        logging.warning("Invalid mon command '%s'", cmd)
-                        resp = 'Invalid Command: "%s"\n' % cmd
-                        resp = hexEncode(resp)
-
-                if self.target.getState() != TARGET_HALTED:
-                    logging.error("Remote command left target running!")
-                    logging.error("Forcing target to halt")
-                    self.target.halt()
-
-            return self.createRSPPacket(resp)
+            return self.handleRemoteCommand(cmd)
 
         else:
             return self.createRSPPacket("")
 
+    # TODO rewrite the remote command handler
+    def handleRemoteCommand(self, cmd):
+        logging.debug('Remote command: %s', cmd)
+
+        safecmd = {
+            'reset' : ['Reset target', 0x1],
+            'halt'  : ['Halt target', 0x2],
+            'resume': ['Resume target', 0x4],
+            'help'  : ['Display this help', 0x80],
+            'reg'   : ['Show registers', 0],
+            'init'  : ['Init reset sequence', 0]
+        }
+        resultMask = 0x00
+        resp = 'OK'
+        if cmd == 'help':
+            for k,v in safecmd.items():
```

Comment:
resp should be set to an empty string before this for loop, otherwise 'OK' will show up in the help text.
```
(gdb) mon help
OKreset Reset target
help    Display this help
resume  Resume target
init    Init reset sequence
reg     Show registers
halt    Halt target
```